Client/Frontend/Shields/ShieldsActivityItemSourceProvider.swift

```diff
@@ -0,0 +1,127 @@
+// Copyright 2020 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import LinkPresentation
+import BraveShared
+import BraveUI
+import Shared
+
+// MARK: - ActivityTypeValue
+
+public enum ActivityTypeValue: String, CaseIterable {
+    case whatsapp = "net.whatsapp.WhatsApp.ShareExtension"
+    case slack = "com.tinyspeck.chatlyio.share"
+    case gmail = "com.google.Gmail.ShareExtension"
+    case instagram = "com.burbn.instagram.shareextension"
+}
+
+// MARK: - ShieldsActivityItemSourceProvider
+
+final class ShieldsActivityItemSourceProvider {
+    
+    static let shared = ShieldsActivityItemSourceProvider()
+    
+    func setupGlobalShieldsActivityController(theme: Theme) -> UIActivityViewController {
+        let backgroundImage = #imageLiteral(resourceName: "share-activity-background")
+        
+        let statsView = UIView(frame: CGRect(size: backgroundImage.size)).then {
+            let backgroundImageView = UIImageView(image: backgroundImage)
+            let statsInfoView = BraveShieldStatsView().then {
+                $0.applyTheme(theme)
+            }
+            
+            $0.addSubview(backgroundImageView)
+            $0.addSubview(statsInfoView)
+            
+            backgroundImageView.snp.makeConstraints {
+                $0.edges.equalToSuperview()
+            }
+            statsInfoView.snp.makeConstraints {
+                $0.centerX.equalToSuperview()
+                $0.centerY.equalToSuperview().offset(15)
+                $0.height.equalToSuperview().multipliedBy(0.40)
+                $0.width.equalToSuperview().multipliedBy(0.70)
+            }
+        }
+    
+        let contentView = UIView(frame: CGRect(width: statsView.frame.width, height: statsView.frame.height + 85)).then {
```

Comment:
What is this 85?

Reply:
We extend the content when the application with that share type doesn't support text and image at the same time. and place the text on the bottom. So this is that extra space to place the text. You can see visually on screenshot in the description.